dbt/adapters/athena/config.py

```diff
@@ -1,9 +1,108 @@
+        self.config = config
+        self.session_kwargs = session_kwargs
+
+    def set_timeout(self) -> int:
+        """
+        Get the timeout value.
+
+        This function retrieves the timeout value from the parsed model's configuration. If the timeout value
+        is not defined, it falls back to the default timeout value. If the retrieved timeout value is less than or
+        equal to 0, a ValueError is raised as timeout must be a positive integer.
+
+        Returns:
+            int: The timeout value in seconds.
+
+        Raises:
+            ValueError: If the timeout value is not a positive integer.
+
+        """
+        timeout = self.config.get("timeout", DEFAULT_SPARK_SESSION_TIMEOUT)
+        if not isinstance(timeout, int):
+            raise TypeError("Timeout must be an integer")
+        if timeout <= 0:
+            raise ValueError("Timeout must be a positive integer")
+        LOGGER.debug(f"Setting timeout: {timeout}")
+        return timeout
+
+    def get_polling_interval(self) -> Any:
+        """
+        Get the polling interval for the configuration.
+
+        Returns:
+            Any: The polling interval value.
+
+        Raises:
+            KeyError: If the polling interval is not found in either `self.config`
+                or `self.session_kwargs`.
+        """
+        try:
+            return self.config["polling_interval"]
+        except KeyError:
+            try:
+                return self.session_kwargs["polling_interval"]
+            except KeyError:
+                return DEFAULT_POLLING_INTERVAL
+
+    def set_polling_interval(self) -> float:
+        """
+        Set the polling interval for the configuration.
+
+        Returns:
+            float: The polling interval value.
+
+        Raises:
+            ValueError: If the polling interval is not a positive integer.
+        """
+        polling_interval = self.get_polling_interval()
+        if not (isinstance(polling_interval, float) or isinstance(polling_interval, int)) or polling_interval <= 0:
+            raise ValueError(f"Polling_interval must be a positive number. Got: {polling_interval}")
+        LOGGER.debug(f"Setting polling_interval: {polling_interval}")
+        return float(polling_interval)
+
+    def set_engine_config(self) -> Dict[str, Any]:
+        """Set the engine configuration.
+
+        Returns:
+            Dict[str, Any]: The engine configuration.
+
+        Raises:
+            TypeError: If the engine configuration is not of type dict.
+            KeyError: If the keys of the engine configuration dictionary do not match the expected format.
+        """
+        engine_config = self.config.get("engine_config", DEFAULT_SPARK_ENGINE_CONFIG)
+        if not isinstance(engine_config, dict):
+            raise TypeError("Engine configuration has to be of type dict")
+
+        expected_keys = {"CoordinatorDpuSize", "MaxConcurrentDpus", "DefaultExecutorDpuSize"}
```

Comment:
all these keys are mandatory? seems so based on this implementation.

Reply:
Yes from the [docs](https://docs.aws.amazon.com/athena/latest/ug/notebooks-spark-getting-started.html#notebooks-spark-getting-started-editing-session-details) it looks like all three are necessary

> Coordinator size - A coordinator is a special executor that orchestrates processing work and manages other executors in a notebook session. Currently, 1 DPU is the default and only possible value.

> Executor size - An executor is the smallest unit of compute that a notebook session can request from Athena. Currently, 1 DPU is the default and only possible value.

> Max concurrent value - The maximum number of DPUs that can run concurrently. The default is 20, the minimum is 3, and the maximum is 60. Increasing this value does not automatically allocate additional resources, but Athena will attempt to allocate up to the maximum specified when the compute load requires it and when resources are available.

Reply:
Hi @Avinash-1394  

> Thank you for this pull request
> I used this branch to do manual executions where trino has limitations like 100 partitions etc.,.
> Can't wait to use it in our production runs

I found that this engine config actually can take more than these arguments and only MaxConcurrentDPUs is the required param: Ref: https://docs.aws.amazon.com/athena/latest/APIReference/API_EngineConfiguration.html

this is the config I tried for iceberg table:

`engine_config={
            "CoordinatorDpuSize": 1, # currently 1 is default and only possible allowed value
            "DefaultExecutorDpuSize": 1, # currently 1 is default and only possible allowed value
            "MaxConcurrentDpus": 60, # The default is 20, the minimum is 3, and the maximum is 60. Increasing this value does not automatically allocate additional resources, but Athena will attempt to allocate up to the maximum specified when the compute load requires it and when resources are available.
            "SparkProperties": {
                "spark.sql.catalog.spark_catalog": "org.apache.iceberg.spark.SparkSessionCatalog",
                "spark.sql.catalog.spark_catalog.catalog-impl": "org.apache.iceberg.aws.glue.GlueCatalog",
                "spark.sql.catalog.spark_catalog.io-impl": "org.apache.iceberg.aws.s3.S3FileIO",
                "spark.sql.extensions": "org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions",
            }
        }`

and here are the UI options for spark notebook:
![image](https://github.com/dbt-athena/dbt-athena/assets/135387788/a6275af0-f028-4982-8b45-6caa8674c168)

Reply:
another sample engine config for spark parameters to enable encryption:
{ 
    "CoordinatorDpuSize": 1, 
    "MaxConcurrentDpus": 20, 
    "DefaultExecutorDpuSize": 1, 
    "SparkProperties": { 
      "spark.authenticate": "true", 
      "spark.io.encryption.enabled": "true", 
      "spark.network.crypto.enabled": "true" 
    } 
}

Reply:
Hey @sankeerthnagapuri,

Thank you so much for providing these examples and validating my branch. Would you be able to share the your model with the entire config block so that I can include it in examples in the README and the MR description? I'm in no way a spark expert so it's so great to see the capabilities being fully explored.

Reply:
Hi @Avinash-1394 

Actually I just got to know that "SparkProperties" is not something available in boto3 version 1.26
If we add this property in engine configuration, it is throwing error that "SparkProperties" is not a valid configuration.
So looks like what you have is valid for this boto3 version.

But this is a must have property if we want to work with non-hive tables like hudi, iceberg or delta lake.
So either we say this release doesnt support non-hive tables or we have to upgrade boto3 version to latest.